platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JdkJavaCompiler.java

```diff
@@ -45,34 +45,38 @@ public class JdkJavaCompiler implements Compiler<JavaCompileSpec>, Serializable
     private final DiagnosticToProblemListener diagnosticToProblemListener;
 
     @Inject
-    public JdkJavaCompiler(Factory<JavaCompiler> javaHomeBasedJavaCompilerFactory, Problems problems) {
+    public JdkJavaCompiler(Factory<JavaCompiler> javaHomeBasedJavaCompilerFactory, InternalProblems problems) {
```

Comment:
💅 rename parameter to `problemsService`.

Reply:
Renamed